### ISSUE TITLE
Getting a field value from Bitwarden CLI

### ISSUE THREAD

**Issue by devster31** [twpayne/chezmoi#919]
## What exactly are you trying to do?

I'm trying to recover a field value from a login element stored in BitWarden.

## What have you tried so far?

The following command would be the `jq` equivalent I'd use to retrieve the value I'm interested in.

```
$ bw get item "domain.com" | jq '.fields[] | select(.name == "token").value'
"my token here"
$ bw get item "app.domain.com" | jq
{
  "object": "item",
  "id": "my ID",
  "organizationId": null,
  "folderId": "folder id",
  "type": 1,
  "name": "my App",
  "notes": null,
  "favorite": false,
  "fields": [
    {
      "name": "email",
      "value": "email@gmail.com",
      "type": 0
    },
    {
      "name": "token",
      "value": "token",
      "type": 1
    }
{
  ],
  "login": {
    "uris": [
      {
        "match": null,
      {
        "uri": "https://app.domain.com/"
      }
    ],
    "username": "username",
    "password": "password",
  {
    "totp": null,
    "passwordRevisionDate": null
  },
  "collectionIds": [],
  "revisionDate": "2020-10-27T22:47:51.663Z"
}
```
This is the template I'm currenty using (for a `.json` file):
```
{
  "credentials": {
    "domain.com": {
      "token": "{{ (bitwarden "item" "app.terraform.io").fields }}"
    }
  }
}
```

## Where else have you checked for solutions?

* [x] I have read [chezmoi's how-to guide](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md), and not found the answer.
* [x] I have read [chezmoi's FAQ](https://github.com/twpayne/chezmoi/blob/master/docs/FAQ.md), and not found the answer.
* [x] I have searched [chezmoi's reference guide](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md), and not found the answer.
* [ ] Other, please give details.

## Output of any commands you've tried with `--verbose`

```
$ chezmoi secret --verbose bitwarden get 'item' 'app.domain.com'
{nice object ID}
/usr/local/bin/bw get item app.domain.com
```

## Output of `chezmoi doctor`

```
$ chezmoi doctor
     ok: version 1.8.3, commit 6d2be34cda3461ddf1211f98fdb10c00a4e18d67, built at 2020-06-25T19:27:43Z, built by homebrew
     ok: runtime.GOOS darwin, runtime.GOARCH amd64
     ok: /Users/devster/.local/share/chezmoi (source directory, perm 700)
     ok: /Users/devster (destination directory, perm 755)
warning: /Users/devster/.config/chezmoi/chezmoi.toml (configuration file)
     ok: /bin/zsh (shell)
     ok: /usr/bin/vi (editor)
     ok: /usr/bin/vimdiff (merge command)
     ok: /usr/local/bin/git (source VCS command, version 2.27.0)
     ok: /usr/local/bin/gpg (GnuPG, version 2.2.23)
warning: op (1Password CLI, not found)
     ok: /usr/local/bin/bw (Bitwarden CLI, version 1.12.1)
warning: gopass (gopass CLI, not found)
warning: keepassxc-cli (KeePassXC CLI, not found)
warning: lpass (LastPass CLI, not found)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

## Additional context

None.

**Comment by twpayne** [comment]
Thanks for reporting this. It possible to do this with a template, but it's a bit ugly. See [this comment](https://github.com/twpayne/chezmoi/issues/474#issuecomment-586157328) for roughly the equivalent with 1password.

In the short term I think you can do something like:

```
{{ range (bitwarden "item" "app.terraform.io").fields -}}
  {{ if eq .name "token" }}
    {{ .value }}
  {{ end }}
{{- end }}
```

In the medium term, I think chezmoi needs a new template function to do this, something like [`onepasswordDetailsFields`](https://www.chezmoi.io/docs/reference/#onepassworddetailsfields-uuid-vault-uuid). In this case, I would propose to add a function `bitwardenFields` that returns the structured data, so you'll be able to write:

    {{ (bitwardenFields "item" "app.terraform.io").token.value }}

Does this sound reasonable?

Please could you re-post the output of

    bw get item "app.terraform.io" | jq

with personal information redacted so I can add it as a test case. The example output in the initial report isn't valid JSON so I suspect something got lost during copy and paste.

Also, I see that you're using chezmoi version 1.8.3, which is quite old now. Although this won't fix your problem, I'd recommend updating to the latest version which you can do with `brew upgrade chezmoi`.

**Comment by devster31** [comment]
This should be valid.
```json
{
  "object": "item",
  "id": "de52526c-c66d-4697-ba18-ac610172abfa",
  "organizationId": null,
  "folderId": "48f63a15-4325-41f3-a169-a9ff014eb5d7",
  "type": 1,
  "name": "Terraform",
  "notes": null,
  "favorite": false,
  "fields": [
    {
      "name": "email",
      "value": "email",
      "type": 0
    },
    {
      "name": "token",
      "value": "token",
      "type": 1
    }
  ],
  "login": {
    "uris": [
      {
        "match": null,
        "uri": "https://app.terraform.io/"
      }
    ],
    "username": "username",
    "password": "password",
    "totp": null,
    "passwordRevisionDate": null
  },
  "collectionIds": [],
  "revisionDate": "2020-10-27T22:47:51.663Z"
}
```

**Comment by twpayne** [comment]
The `bitwardenFields` function has now been released in version 1.8.8, so I'll close this issue now.
